python/tidy/servo_tidy/tidy.py

```diff
@@ -723,13 +724,40 @@ def parse_config(content):
     config['ignore']['directories'] = map(lambda path: os.path.join(*path.split('/')),
                                           config['ignore']['directories'])
 
+    # Add dict of dir, list of expected ext to config
+    dirs_to_check = config_file.get("check_ext", {})
+    # Fix the paths (OS-dependent)
+    for path, exts in dirs_to_check.items():
+        fixed_path = os.path.join(*path.split('/'))
+        config['check_ext'][fixed_path] = exts
+
     # Override default configs
     user_configs = config_file.get("configs", [])
     for pref in user_configs:
         if pref in config:
             config[pref] = user_configs[pref]
 
 
+def check_directory_files(directories, print_text=True):
+    if print_text:
+        print '\rChecking directories for correct file extensions...'
+    for directory, file_extensions in directories.items():
+        files = os.listdir(directory)
+        for filename in files:
+            full_path = os.path.join(directory, filename)
+            if not os.path.isfile(full_path):
+                continue
```

Comment:
Ok so I removed the entire `isfile` check, if necessary we could always add a 'allow_subdir' option or something, for now we could just keep it focused on checking file extensions and assume the dirs will not have sub-dirs(I checked and now it actually sees a sub-dir as a file with the wrong extension...).

I also removed the test case using the `empty` dir, which off course never got added to the repo since it's empty and we don't need it anyway, if the dir is empty the function will simply not check anything since the iterable it operates on will be empty too...